src/DripsDeployer.sol

```diff
@@ -0,0 +1,412 @@
+        }
+    }
+
+    function moduleSalts() public view returns (bytes32[] memory) {
+        return _moduleSalts;
+    }
+
+    function moduleAddress(bytes32 salt) public view returns (address addr) {
+        return Create3.computeAddress(salt);
+    }
+}
+
+abstract contract BaseModule {
+    address public immutable dripsDeployer;
+    bytes32 public immutable moduleSalt;
+
+    constructor(address dripsDeployer_, bytes32 moduleSalt_) {
+        require(
+            address(this) == Create3.computeAddress(moduleSalt_, dripsDeployer_),
+            "Invalid module deployment salt"
+        );
+        dripsDeployer = dripsDeployer_;
+        moduleSalt = moduleSalt_;
+    }
+
+    function args() public view virtual returns (bytes memory);
+
+    function _moduleAddress(bytes32 salt) internal view returns (address addr) {
+        return Create3.computeAddress(salt, dripsDeployer);
+    }
+
+    modifier onlyModule(bytes32 salt) {
+        require(msg.sender == _moduleAddress(bytes32(salt)));
+        _;
+    }
+}
+
+abstract contract ContractDeployerModule is BaseModule {
+    bytes32 public immutable salt = "deployment";
+
+    function deployment() public view returns (address) {
+        return Create3.computeAddress(salt);
+    }
+
+    function deploymentArgs() public view virtual returns (bytes memory);
+
+    function _deployContract(bytes memory creationCode) internal {
+        Create3.deploy(0, salt, abi.encodePacked(creationCode, deploymentArgs()));
+    }
+}
+
+abstract contract ProxyDeployerModule is BaseModule {
+    bytes32 public immutable proxySalt = "proxy";
+    address public proxyAdmin;
+    address public logic;
+
+    function proxy() public view returns (address) {
+        return Create3.computeAddress(proxySalt);
+    }
+
+    function proxyArgs() public view returns (bytes memory) {
+        return abi.encode(logic, proxyAdmin);
+    }
+
+    function logicArgs() public view virtual returns (bytes memory);
+
+    // slither-disable-next-line reentrancy-benign
+    function _deployProxy(address proxyAdmin_, bytes memory logicCreationCode) internal {
+        // Deploy logic
+        address logic_;
+        bytes memory logicInitCode = abi.encodePacked(logicCreationCode, logicArgs());
+        // slither-disable-next-line assembly
+        assembly ("memory-safe") {
+            logic_ := create(0, add(logicInitCode, 32), mload(logicInitCode))
+        }
+        require(logic_ != address(0), "Logic deployment failed");
+        logic = logic_;
+        // Deploy proxy
+        proxyAdmin = proxyAdmin_;
+        // slither-disable-next-line too-many-digits
+        bytes memory proxyInitCode = abi.encodePacked(type(ManagedProxy).creationCode, proxyArgs());
+        Create3.deploy(0, proxySalt, proxyInitCode);
+    }
+}
+
+abstract contract DripsDependentModule is BaseModule {
+    // slither-disable-next-line naming-convention
+    bytes32 internal immutable _dripsModuleSalt = "Drips";
```

Comment:
Maybe always with a pre-fix?
```suggestion
    bytes32 internal immutable _dripsModuleSalt = "drips-contracts.drips";
```

Reply:
In a way it already is prefixed, the default seed for the `DripsDeployer` in the singleton factory is `DripsDeployer`, so it's already in the address deployment chain. I don't think that we want to deploy other protocols using that factory, so prefixing it is probably an overkill. Having the salt matching the contract name simplifies the deployment script too.